skyportal/app_server.py

```diff
@@ -111,13 +111,19 @@ def log_request(self, handler):
     (r'/api/candidates(/[0-9A-Za-z-_]+)/([0-9]+)', CandidateHandler),
     (r'/api/candidates(/.*)?', CandidateHandler),
     (r'/api/classification(/[0-9]+)?', ClassificationHandler),
-    (r'/api/comment', CommentHandler),
-    (r'/api/comment(/[0-9]+)/attachment', CommentAttachmentHandler),
+    (
+        r'/api/comment(/[0-9]+)/attachment(/(?:object|spectrum))?',
+        CommentAttachmentHandler,
+    ),
     # Allow the '.pdf' suffix for the attachment route, as the
     # react-file-previewer package expects URLs ending with '.pdf' to
     # load PDF files.
-    (r'/api/comment(/[0-9]+)/attachment.pdf', CommentAttachmentHandler),
-    (r'/api/comment(/[0-9]+)?(/.+)?', CommentHandler),
+    (
+        r'/api/comment(/[0-9]+)/attachment.pdf(/(?:object|spectrum))?',
+        CommentAttachmentHandler,
+    ),
```

Comment:
Does the ?: in the parenthesis do anything in this context or is it just faster to match things?

Reply:
The `?:` means it's a non-matching group.  I needed to specify `(object|spectrum)`, which I'm not sure how to do without brackets.  The `?:` is to say: don't try and *use* those brackets.  I *think* I checked and saw that removing that breaks things, but may be worth double checking.